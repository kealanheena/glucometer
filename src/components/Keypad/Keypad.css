.keypad-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  height: var(--keypad-height);
  padding: 4%;
  margin: 4%;
  width: var(--keypad-width);
}

.button-container {
  display: flex;
  flex-direction: row;
  height: 20%;
  width: 100%;
}
