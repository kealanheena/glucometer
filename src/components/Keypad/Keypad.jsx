import React from 'react';
import PropTypes from 'prop-types';
import Key from '../Key/Key';
import './Keypad.css';

const Keypad = ({ 
  buttons, 
  calculateBolus,
}) => {

  // const buttonKeys = buttons.map(button => <p key={button}>{button}</p>);

  return (
    <div className="keypad-container">
      <div className="button-key">
        <Key 
          keyAction={calculateBolus} 
          keyValue='Calculate Bolus'
        />
      </div>
    </div>
  );
}

Keypad.propTypes = {
  buttons: PropTypes.array.isRequired,
}

export default Keypad;