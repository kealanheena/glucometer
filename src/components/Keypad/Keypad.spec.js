import React from 'react';
import { shallow } from 'enzyme';
import Keypad from './Keypad';

describe('Keypad', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Keypad buttons={[]}/>
    )
  });

  it('should render 2 <div/>\'s', () => {
    expect(wrapper.find('div').length).toEqual(2);
  });

  it('renders the values of the buttons', () => {
    wrapper.setProps({buttons: ['test_one', 'test_two']});
    expect(wrapper.find('.buttons-container').text()).toEqual('test_onetest_two');
  });
});