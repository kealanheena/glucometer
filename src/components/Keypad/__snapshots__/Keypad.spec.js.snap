// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
<div
  className="keypad-container"
>
  <div
    className="button-key"
  >
    <Key
      keyValue="Calculate Bolus"
    />
  </div>
</div>
`;
